### PR TITLE
Multilayer example

### PR DESCRIPTION
**Pull request by mdenil**
The indexing error in data generation was bugging me.

I also extended the RNN example to show how to do more than one layer.

### CODE REVIEW

**Comment by bshillingford** [comment]
Nice, a multi-layer RNN is a good example.  I'm hoping to get a chance over the weekend to add in some better docs, btw.

File: examples/rnn_example.lua [hunk 1]
```diff
@@ -10,48 +10,66 @@ local n_hidden = 25
...
+
+        return loss
+    end,
+
+    backward = function(self)
+        self.tape:backward()
+    end,
+
+    get_parameters = function(self)
+        local pack = nn.Sequential()
```
Comment (bshillingford) [inline examples/rnn_example.lua:66]: Not sure if I'm being picky, but do you think it's a bit misleading to use an nn.Sequential when things aren't sequential? I think an instance of nn.Container would be more clearly a dummy object, especially for a tutorial-style example.

Comment (mdenil) [inline examples/rnn_example.lua:66]: Yes that's a good idea. I just default to nn.Sequential() out of habit.

**Comment by mdenil** [comment]
A non-trivial LSTM example would be cool.  We could borrow Kaparthy's tinyshakespear data and do a language model demo.

**Comment by bshillingford** [comment]
Yeah, I just saw his blog post, lots of neat demos there. That or any of
the other data would be a good example.

**Pull request merged**